File: Summary Ranges - Leetcode 228.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        ans = []     
        i = 0 
        
        while i < len(nums): 
            start = nums[i]  
            while i < len(nums)-1 and nums[i] + 1 == nums[i + 1]: 
                i += 1 
            
            if start != nums[i]: 
                ans.append(str(start) + "->" + str(nums[i]))
            else: 
                ans.append(str(nums[i]))
            
            i += 1

        return ans
        # Time: O(n)
        # Space: O(n)

File: Summary Ranges - Leetcode 228/Summary Ranges - Leetcode 228.go
package solution

func summaryRanges(nums []int) []string {
	res := []string{}
    if(len(nums)==0){
        return res
    }
	i, j := 0, 0
	for i < len(nums)-1 {
		if nums[i] != nums[i+1]-1 {
			if i == j {
				res = append(res, strconv.Itoa(nums[i]))
			} else {
				res = append(res, fmt.Sprintf("%d->%d", nums[j], nums[i]))
			}
			j = i + 1
		}
		i++
	}
	if i == j {
		res = append(res, strconv.Itoa(nums[i]))
	} else {
		res = append(res, fmt.Sprintf("%d->%d", nums[j], nums[i]))
	}
	return res
}
